### PR TITLE
[FE] Implement Tag and Question References for Answer Creation

### PR DESCRIPTION
**Pull request by asligook**
## 📋 Proposed Changes

  - The custom React Markdown component will allow users to mention tags and questions while creating answers.

## 🧪 Included Tests
- [X] The user follows the writing format for tags : [tag name](#tag-105), questions: [question title](# q-2)
- [X] The user clicks on the link and navigates to the related tag or question page.

### CODE REVIEW

File: frontend/src/components/CreateAnswerForm.tsx [hunk 1]
```diff
@@ -112,7 +114,18 @@ export function CreateAnswerForm({ questionId }: CreateAnswerFormProps) {
 
       {isPreviewMode ? (
         <div className="min-h-[200px] rounded-lg border border-gray-300 bg-white p-4">
-          <ContentWithSnippets content={content} />
```
Comment (mmtftr) [inline frontend/src/components/CreateAnswerForm.tsx:115]: Don't add a new ReactMarkdown here, you should add the custom component to the ReactMarkdown element within ContentWithSnippets

**Review by mmtftr — CHANGES_REQUESTED** [review]

File: frontend/src/components/CreateAnswerForm.tsx (hunk 1 above)
Comment (asligook) [reply frontend/src/components/CreateAnswerForm.tsx:115]: That was the most confusing part, thanks for the tip!  Working on it.

**Review by mmtftr — APPROVED** [review]
Thanks a lot, looks good to me

**Comment by asligook** [comment]
Closes #624.

**Pull request merged**